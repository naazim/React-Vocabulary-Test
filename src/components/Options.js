import React from 'react';
import Option from './Option';

const Options = (props) => (
  <div>
    <div className="widget__header">
      <h3 className="widget__title">Your word list</h3>
      <button
        className="button--link"
        onClick={props.handleDeleteOptions}
      >
        Remove All
    </button>
    </div>

    {props.options.length === 0 && <p className="widget__message">Please create Vocabulary list to get started!</p>}
    {
      props.options.map((option, index) => {
        return (
          <Option
            key={index}
            optionText={option.native}
            optionText2={option.foreign}
            count={index + 1}
            handleDeleteOption={props.handleDeleteOption}
          />
        )
      })
    }
  </div>
);

export default Options;
