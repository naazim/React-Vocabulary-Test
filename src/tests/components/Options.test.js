import React from 'react';
import { shallow } from 'enzyme';
import Options from '../../components/Options';

describe('Options', () => {
    const handleDeleteAll = jest.fn();
    let options= [{"native":"Hello","foreign":"Hallo"},{"native":"Good","foreign":"Gut"}];
    let wrapper = shallow(<Options options={options} handleDeleteAll={handleDeleteAll} />);

    test('should render options correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should delete entire list on click of remove all mlink', () => {
        wrapper.find('.button--link').simulate('click');
        expect(handleDeleteAll).toHaveBeenCalled();
    });

    test('should render an option for every object in the word list array', () => {
        expect(wrapper.find('Option').length).toBe(2);
    });

    test('should show message to add words when list is empty', () => {
        expect(wrapper.find('.widget__message').length).toBe(0);
        options= [];
        wrapper = shallow(<Options options={options} handleDeleteAll={handleDeleteAll} />);
        expect(wrapper.find('.widget__message').length).toBe(1);
        expect(wrapper.find('.widget__message').text()).toBe('Enter 20 words with translations get started');
        expect(wrapper).toMatchSnapshot();
    });

});