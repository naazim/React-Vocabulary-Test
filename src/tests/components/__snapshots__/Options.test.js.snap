// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should render options correctly 1`] = `
<div>
  <div
    className="widget__header"
  >
    <h3
      className="widget__title"
    >
      Your word list
    </h3>
    <button
      className="button--link"
      onClick={[Function]}
    >
      Remove All
    </button>
  </div>
  <Option
    count={1}
    foreignWord="Hallo"
    key="0"
    nativeWord="Hello"
  />
  <Option
    count={2}
    foreignWord="Gut"
    key="1"
    nativeWord="Good"
  />
</div>
`;

exports[`Options should show message to add words when list is empty 1`] = `
<div>
  <div
    className="widget__header"
  >
    <h3
      className="widget__title"
    >
      Your word list
    </h3>
    <button
      className="button--link"
      onClick={[Function]}
    >
      Remove All
    </button>
  </div>
  <p
    className="widget__message"
  >
    Enter 20 words with translations get started
  </p>
</div>
`;
